src/Microsoft.ML.PCA/PcaTransform.cs

```diff
@@ -554,12 +669,131 @@ private static void TransformFeatures(IExceptionContext ectx, ref VBuffer<Float>
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            foreach (var colInfo in _columns)
+            {
+                if (!inputSchema.TryFindColumn(colInfo.Input, out var col))
+                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.Input);
+
+                if (col.Kind != SchemaShape.Column.VectorKind.Vector || col.ItemType.RawKind != DataKind.R4)
+                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.Input);
+
+                result[colInfo.Output] = new SchemaShape.Column(colInfo.Output,
+                    SchemaShape.Column.VectorKind.Vector, NumberType.R4, false);
+            }
+
+            return new SchemaShape(result.Values);
+        }
+    }
+
+    public static class PcaEstimatorExtensions
+    {
+        private sealed class OutPipelineColumn : Vector<float>
+        {
+            public readonly Vector<float> Input;
+
+            public OutPipelineColumn(Vector<float> input, string weightColumn, int rank,
+                                     int overSampling, bool center, int? seed = null)
+                : base(new Reconciler(weightColumn, rank, overSampling, center, seed), input)
+            {
+                Input = input;
+            }
+        }
+
+        private sealed class Reconciler : EstimatorReconciler
+        {
+            private readonly PcaTransform.ColumnInfo _colInfo;
+
+            public Reconciler(string weightColumn, int rank, int overSampling, bool center, int? seed = null)
+            {
+                _colInfo = new PcaTransform.ColumnInfo(
+                    null, null, weightColumn, rank, overSampling, center, seed);
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                // Only one column is allowed.
+                Contracts.Assert(toOutput.Length == 1);
+                var outCol = (OutPipelineColumn)toOutput[0];
+                var inputColName = inputNames[outCol.Input];
+                var outputColName = outputNames[outCol];
+                return new PcaEstimator(env, inputColName, outputColName,
+                                         _colInfo.WeightColumn, _colInfo.Rank, _colInfo.Oversampling,
+                                         _colInfo.Center, _colInfo.Seed);
+            }
+        }
+
+        /// <summary>Compute the principal components of the input column. Can significantly reduce size of vector.</summary>
```

Comment:
>Compute the principal components of the input column. [](start = 21, length = 53)

describe the transformation, not the training procedure. 'Replaces vector with its projection to the principal component subspace' or something along these lines

Reply:
updated

---
In reply to: [227594833](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227594833) [](ancestors = 227594833)